R/gen_msf_data.R

```diff
@@ -857,6 +857,123 @@ gen_msf_data <- function(dictionary, dat_dict, is_survey, varnames = "data_eleme
+
+    ## date triage
+    dis_output <- enforce_timing(dis_output,
+                                 first  = "date_triage",
+                                 second = "date_verification",
+                                 5:30,
+                                 inclusive = TRUE
+    )
+
+    # date verification
+    dis_output <- enforce_timing(dis_output,
+                                 first  = "date_verification",
+                                 second = "date_assessment",
+                                 5:30
+    )
+
+    ## date assessment
+    dis_output <- enforce_timing(dis_output,
+                                 first  = "date_assessment",
+                                 second = "date_response_started",
+                                 5:30,
+                                 inclusive = TRUE
+    )
+
+    ## date response started
+    dis_output <- enforce_timing(dis_output,
+                                 first  = "date_response_started",
+                                 second = "date_response_ended",
+                                 5:30,
+                                 inclusive = TRUE
+    )
+
+    ## create an initials variable of 3 random letters
+    dis_output$initials <- stringi::stri_rand_strings(nrow(dis_output), 2,
+                                                      pattern = "[a-z]")
+
+
+    ## create a signal id
+    dis_output$signal_id <- paste0(dis_output$initials, "_",
+                                      dis_output$location_signal, "_",
+                                      dis_output$signal_type, "_",
+                                      dis_output$date_signal)
+
+
+    ## Copy this signal id to assessment signal id variable
+    dis_output$c_signal_id <- dis_output$signal_id
+
+    ## Copy this signal id to response signal id variable
+    dis_output$c_signal_id2 <- dis_output$signal_id
+
+
+    ## Add a random  number for total people affected
+    dis_output$total_affected <- sample(1:10, nrow(dis_output), replace = TRUE)
+
+
+    ## Add one person less for under 5 affected
+    dis_output$under5_affected <- dis_output$total_affected - 1
+
+
+    ## Add a random  number for total people identified by active case finding
+    dis_output$acf_total <- sample(2:15, nrow(dis_output), replace = TRUE)
+
+    ## Add one person less for total under 5 identified by active case finding
+    dis_output$acf_under5 <- dis_output$acf_total - 1
+
+
+    ## Ensure that alert_status takes a value of 1 where risk characterisation
+    ## is y (yes)
+    dis_output$alert_status[dis_output$risk_characterisation == "y"] <- 1
+
+
+    ## Ensure that alert_status takes a value of 0 where risk characterisation
+    ## is n (n) or u (unsure)
+    dis_output$alert_status[dis_output$risk_characterisation == "n" |
+                              dis_output$risk_characterisation == "u"] <- 0
+
+    ## Convert all columns NA when no intervention required
+    dis_output[dis_output$alert_status == 0, 42:49] <- NA
```

Comment:
please write out var names (below too) - this is not a safe way of ensuring this selects correct columns going forward

Reply:
Should I mention all of them, or just the start name and end name?

Reply:
would say ideally just list all of them - then we dont end up in a headache down the line where the dataset changes order and we need to scratch our heads figuring out which variables we meant to use

Reply:
ah actually just realised how many variabels that is - you know what just leave it and we can deal with the fallout...